.github/workflows/regenerate-code.yml

```diff
@@ -0,0 +1,39 @@
+name: Regenerate all code
+on: workflow_dispatch
+
+jobs:
+  build:
+    timeout-minutes: 120
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup node 18
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18'
+
+      - name: Setup .net core sdk
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: '7.0.x'
+
+      - name: Install packages
+        run: |
+          npm ci
+
+      - name: Generate code
+        shell: pwsh
+        run: |
+          ./eng/Generate.ps1
+
+      - name: Commit generated code
+        run: |
+          git config --global user.name 'actions-user'
+          git config --global user.email 'action@github.com'
+          git add .
+          git commit -m "re-generate test code"
```

Comment:
maybe we could add the `allow-empty` option here to ensure this command will always succeed and we always see a new commit by github action to distinguish the case that this feature is in a malfunction

Reply:
Updated. It will be like
![image](https://github.com/Azure/autorest.csharp/assets/97225342/68379088-f4c4-42eb-b1af-ebc4ccbfef9b)